splitgraph/cloud/tunnel_client.py

```diff
@@ -24,6 +29,45 @@
 
 """
 
+# Download URL and MD5 hash of rathole build ZIP archive for each supported platform.
+RATHOLE_BUILDS = {
+    "Darwin": (
+        "https://github.com/rapiz1/rathole/releases/download/v0.4.4/rathole-x86_64-apple-darwin.zip",
+        "8d21125f785505a72497565a20288226",
+    ),
+    "Windows": (
+        "https://github.com/rapiz1/rathole/releases/download/v0.4.4/rathole-x86_64-pc-windows-msvc.zip",
+        "89453d34db16fcdda64719c819c81cf2",
+    ),
+    "Linux-x86_64-glibc": (
+        "https://github.com/rapiz1/rathole/releases/download/v0.4.4/rathole-x86_64-unknown-linux-gnu.zip",
+        "ec9a8a0b371adac2d8dd33dc7a5fa2ab",
+    ),
+    "Linux-x86_64-musl": (
+        "https://github.com/rapiz1/rathole/releases/download/v0.4.4/rathole-x86_64-unknown-linux-musl.zip",
+        "4fa7586925a7f28e8dbf225c5790e5d3",
+    ),
+}
+
+
+def get_md5(stream):
+    return hashlib.md5(stream.read()).hexdigest()
```

Comment:
Do you need to `stream.seek(0)` here to rewind the stream (since `stream.read()` will consume it)?

Reply:
Nice catch! It works as is without the `seek()`, but only because `ZipFile` does an [implicit `seek()`](https://github.com/python/cpython/blob/88a3f1873ec4e6d35b1736c480c7087e01fca40f/Lib/zipfile.py#L264). I'll add the explicit `seek` to make it more obvious we're rewinding the tape! Thanks for catching that!